api/src/main/java/org/openmrs/api/handler/PatientDataUnvoidHandler.java

```diff
@@ -72,6 +73,11 @@ public void handle(Patient patient, User originalVoidingUser, Date origParentVoi
 					}
 				}
 			}
+
+			if (!patient.getVoided()) {
+				CohortService cs = Context.getCohortService();
+				cs.patientVoided(patient);
```

Comment:
This is incorrect, the unvoid handler should unvoid any memberships of the patient whose endDate matches the patient's dateVoided